lib/src/models/gateway/events/voice.dart

```diff
@@ -38,3 +42,63 @@ class VoiceServerUpdateEvent extends DispatchEvent {
   /// The guild.
   PartialGuild get guild => gateway.client.guilds[guildId];
 }
+
+/// {@template voice_channel_effect_send_event}
+/// Emitted when someone sends an effect, such as an emoji reaction or a soundboard sound, in a voice channel the current user is connected to.
+/// {@endtemplate}
+class VoiceChannelEffectSendEvent extends DispatchEvent {
+  /// The ID of the channel this effect was sent in.
+  final Snowflake channelId;
+
+  /// The ID of the guild this effect was sent in.
+  final Snowflake guildId;
+
+  /// The ID of the user who sent this effect.
+  final Snowflake userId;
+
+  /// The emoji sent, for emoji reaction and soundboard effects.
+  final Emoji? emoji;
+
+  /// The type of emoji animation, for emoji reaction and soundboard effects.
+  final AnimationType? animationType;
+
+  /// The ID of the emoji animation, for emoji reaction and soundboard effects.
+  final int? animationId;
+
+  /// The ID of the soundboard sound, for soundboard effects.
+  final Snowflake? soundId;
+
+  /// The volume of the soundboard sound, from 0 to 1, for soundboard effects.
+  final double? soundVolume;
+
+  /// {@macro voice_channel_effect_send_event}
+  /// @nodoc
+  VoiceChannelEffectSendEvent({
+    required super.gateway,
+    required this.channelId,
+    required this.guildId,
+    required this.userId,
+    required this.emoji,
+    required this.animationType,
+    required this.animationId,
+    required this.soundId,
+    required this.soundVolume,
+  });
+
+  /// The channel this effect was sent in.
+  VoiceChannel get channel => gateway.client.channels[channelId] as VoiceChannel;
```

Comment:
Will error; this should return `PartialChannel`. We currently don't have a way to specify the type of a partial channel (even `PartialTextChannel` only implements `SnowflakeEntity<Channel>`).